awx/api/serializers.py

```diff
@@ -4816,7 +4818,7 @@ def check_credential_permission(self, request, requested_use_credentials):
             raise serializers.ValidationError(_(f"Credentials {not_allowed} not found or you don't have permissions to access it"))
 
     def check_label_permission(self, requested_use_labels):
-        accessible_use_labels = {tup.id for tup in Label.objects.all()}
+        accessible_use_labels = {tup.id for tup in Label.objects.all(organization__in=Organizations.accessible_pk_qs(request.user, 'read_role'))}
```

Comment:
this filters down to labels with organizations the user has read access to, which matches what execution environments use